crates/turborepo-ui/src/color_selector.rs

```diff
@@ -65,13 +74,40 @@ impl ColorSelectorInner {
 
     fn insert_color(&mut self, key: String) -> &'static Style {
         let colors = get_terminal_package_colors();
-        let chosen_color = &colors[self.idx % colors.len()];
+        let chosen_color = Self::get_color_id_by_key(self, &key);
         // A color might have been chosen by the time we get to inserting
-        self.cache.entry(key).or_insert_with(|| {
-            // If a color hasn't been chosen, then we increment the index
-            self.idx += 1;
-            chosen_color
-        })
+        self.cache
+            .entry(key)
+            .or_insert_with(|| &colors[chosen_color])
+    }
+
+    pub fn get_color_hash_by_key(key: &str) -> u64 {
+        let mut hasher = DefaultHasher::new();
+        key.hash(&mut hasher);
+        hasher.finish()
+    }
+
+    fn get_color_id_by_key(&mut self, key: &str) -> usize {
```

Comment:
This implementation suffers from the same issue as the existing one where adding/removing tasks will result in a changed color for the task when there are hash collisions. If both `a#build` and `b#build` get the same initial `color_id`, then removing `a#build` will result in `b#build` changing colors.

Given we're targeting such a small hash (5 at the moment, 256 if we allow for all colors) hash collisions are fairly likely.